### DISCUSSION TITLE
How can we check if the error is an ZodError?

### DISCUSSION THREAD

**Discussion by andredewaard** [colinhacks/zod#2415] [Q&A] [answered]
I try to write a function that transforms errors from Zod or from my backend into a readable format. But to do this i need to know if the error thrown is from Zod or my api. 
To check if its Zod i try to do what is mentioned here: https://github.com/colinhacks/zod/issues/515
Which works as long as i use this during development. When i bundle this script with Vite into a library and use it on a project is fails to check if the error is coming from Zod. 

Is there any reliable way to check this?

**Comment by andredewaard** [comment]
My temp fix for this right now it this:
```
error instanceof Error && 'issues' in error
```
But i feel like this is really error prone.

**Reply by FoodlyEric** [reply, marked as answer]
import {ZodError} from 'zod'

error instanceof ZodError

**Reply by MikeDabrowski** [reply]
![image](https://github.com/colinhacks/zod/assets/17836911/92e3b8ed-a28b-4204-a3e8-46bf6e866b7a)


What am I doing wrong ?

**Reply by Depender-Sethi** [reply]
Try this

```ts
import { ZodError } from 'zod';

export function isZodError(err: unknown): err is ZodError {
  return Boolean(
    err && (err instanceof ZodError || (err as ZodError).name === 'ZodError'),
  );
}
```

**Reply by xieyuheng** [reply]
> import {ZodError} from 'zod'
> 
> error instanceof ZodError

this does not work when many version of zod is used.

**Reply by xieyuheng** [reply]
> Try this
> 
> ```ts
> import { ZodError } from 'zod';
> 
> export function isZodError(err: unknown): err is ZodError {
>   return Boolean(
>     err && (err instanceof ZodError || (err as ZodError).name === 'ZodError'),
>   );
> }
> ```

this does not work when code is minified (bundled).

**Comment by xieyuheng** [comment]
created a pr for this problem: https://github.com/colinhacks/zod/pull/3819
